src/screens/MainApp/MyDiagnoses/index.js

```diff
@@ -0,0 +1,81 @@
+import React, { useState, useEffect } from 'react'
+import { View, FlatList, ActivityIndicator } from 'react-native'
+import { createStackNavigator } from 'react-navigation-stack'
+import { ForceRerenderOnNavigation } from '~/navigationService'
+import * as DatabaseService from '~/databaseService'
+import { sortDiagnosesByMostRecentCreation } from '~/mixins/diagnose'
+import { renderDiagnoses } from './renderUtilities'
+import NoDiagnoses from './NoDiagnoses'
+import styles from './styles'
+
+const MyDiagnoses = () => {
+  const [diagnoses, setDiagnoses] = useState(null)
+  const [downloadingDiagnoses, setDownloadingDiagnoses] = useState(null)
+  const [refetchTrigger, toggleRefetch] = useState(false)
+  const [showNoDiagnosesScreen, setShowNoDiagnosesScreen] = useState(false)
+
+  const refetchDiagnoses = () => {
+    // Tab navigator doesn't unmount components, we need this to refetch need data when pressing this tab.
+    // alternative is to add a query snapshot listener and a MobX store
+    toggleRefetch(!refetchTrigger)
+  }
```

Comment:
I really do not like this. I believe it's ok for now and it really doesn't incur in more database hits (there is a window of 30 minutes of free hits if the data did not change, I could not find this in the docs though, so I ping @nahudalla for insight) But it seems hacky to force a re-render this way.

Reply:
Maybe this will solve the problem: https://reactnavigation.org/docs/en/next/use-focus-effect.html

_We're using React Navigation, right?_

Reply:
I will give it a shot